star_frame_cli/src/template/states_rs

```diff
@@ -17,13 +17,11 @@ pub struct CounterAccount {
 
 pub struct Authority(pub Pubkey);
 
-impl AccountValidate<Authority> for CounterAccount {
-    fn validate_account(self_ref: &Self::Ref<'_>, arg: Authority) -> Result<()> {
-        ensure!(
-            arg.0 == self_ref.authority,
-            "Incorrect authority",
-            {name_pascalcase}Error::IncorrectAuthority
-        );
+impl AccountValidate<&Pubkey> for CounterAccount {
+    fn validate_account(self_ref: &Self::Ref<'_>, signer: &Pubkey) -> Result<()> {
+        if self_ref.authority.ne(signer) {
+            return Err({name_pascalcase}Error::IncorrectAuthority.into());
+        }
```

Comment:
```suggestion
        ensure_eq!(&self_ref.authority, signer, Error::IncorrectAuthority);
```

I think we can still use our ensures here! Our error system has support for that. I must have updated the template wrong when getting that change in!